pvlib/solarposition.py

```diff
@@ -1391,7 +1396,17 @@ def hour_angle(times, longitude, equation_of_time):
         times = times.tz_localize('utc')
     tzs = np.array([ts.utcoffset().total_seconds() for ts in times]) / 3600
 
-    hrs_minus_tzs = (times - times.normalize()) / pd.Timedelta('1h') - tzs
+    # Some timezones have a DST shift at midnight:
+    #   11:59pm -> 1:00am - results in a nonexistent midnight
+    #   12:59am -> 12:00am - results in an ambiguous midnight
+    # We remove the timezone before normalizing for this reason.
+    naive_normalized_times = times.tz_localize(None).normalize()
+
+    # Use Pandas functionality for shifting nonexistent times forward
+    normalized_times = naive_normalized_times.tz_localize(
+        times.tz, nonexistent='shift_forward', ambiguous='raise')
```

Comment:
What about this with the modification `def hour_angle(times, longitude, equation_of_time)` into `def hour_angle(times, longitude, equation_of_time, nonexistent='shift_forward', ambiguous='raise')` to give users a choice to control the `tz_localize` behavior as they want rather than just receiving an error and being blocked?
```suggestion
    # Use Pandas functionality for shifting nonexistent times forward
    normalized_times = naive_normalized_times.tz_localize(
        times.tz, nonexistent=nonexistent, ambiguous=ambiguous)
```

Reply:
@scttnlsn could you comment on this?

Reply:
The user would not necessarily be blocked if the exception is raised.  They'd just need to convert the given times to a non-DST timezone ahead of calling `hour_angle`.  The documentation is updated to mention this.

`infer` won't really work as mentioned here: https://github.com/pvlib/pvlib-python/pull/2133#issuecomment-2233579930  The other options are to `raise` (current approach) or to have `NaN`s returned where the times are ambiguous (the user still likely needs to deal with that case).

I'm personally of the opinion that forcing the user to convert to a non-DST timezone is a fine approach but I could understand someone might want to have `NaN`s instead.